File: leetcode/C++/range-sum-query-immutable.cpp
class NumArray {
public:
    vector<int> v;
    NumArray(vector<int>& nums) {
        for(int &n : nums){
            if(v.empty()) v.push_back(n);
            else v.push_back(v.back()+n);
        }
    }
    
    int sumRange(int left, int right) {
        return (left>0)? v[right]-v[left-1] : v[right];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */

File: leetcode/Python/range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.v = list()
        for n in nums:
            if not self.v:
                self.v.append(n)
            else:
                self.v.append(self.v[-1]+n)

    def sumRange(self, left: int, right: int) -> int:
        return self.v[right] if left==0 else self.v[right]-self.v[left-1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)